packages/next/src/server/config.ts

```diff
@@ -673,6 +673,40 @@ function assignDefaults(
     }
   }
 
+  const userProvidedModularizeImports = result.modularizeImports
+  // Unfortunately these packages end up re-exporting 10600 modules, for example: https://unpkg.com/browse/@mui/icons-material@5.11.16/esm/index.js.
+  // Leveraging modularizeImports tremendously reduces compile times for these.
+  result.modularizeImports = {
+    ...(userProvidedModularizeImports || {}),
+    // This is intentionally added after the user-provided modularizeImports config.
+    '@mui/icons-material': {
+      transform: '@mui/icons-material/{{member}}',
+    },
+    '@mui/material': {
+      transform: '@mui/material/{{member}}',
+    },
+    'date-fns': {
+      transform: 'date-fns/{{member}}',
+    },
+    lodash: {
+      transform: 'lodash/{{member}}',
+    },
+    'lodash-es': {
+      transform: 'lodash-es/{{member}}',
+    },
+    // TODO: Enable this once we have a way to remove the "-icon" suffix from the import path.
+    // Related discussion: https://github.com/vercel/next.js/pull/50900#discussion_r1239656782
+    // 'lucide-react': {
+    //   transform: 'lucide-react/dist/esm/icons/{{ kebabCase member }}',
+    // },
+    ramda: {
+      transform: 'ramda/es/{{member}}',
+    },
+    'react-bootstrap': {
+      transform: 'react-bootstrap/{{member}}',
```

Comment:
This change results in the following error
```bash
Module not found: Can't resolve 'react-bootstrap/useAccordionButton'
  3 | import { Inter } from 'next/font/google'
  4 | import styles from '@/styles/Home.module.css'
> 5 | import { useAccordionButton } from 'react-bootstrap'
  6 | 
  7 | const inter = Inter({ subsets: ['latin'] })
  8 | 
  ```

Reply:
Because some hooks do not have independent files, the above problems are caused. 

If someone encounters the above issue when using react-bootstrap and receives the error message `Module not found: Can’t resolve 'react-bootstrap/xxx'`,
you can solve the problem with the following configuration:

```js
const nextConfig = {
  webpack: config => {
    let modularizeImports = null
    config.module.rules.some(rule => (
      rule.oneOf?.some(oneOf => {
        modularizeImports = oneOf?.use?.options?.nextConfig?.modularizeImports
        return modularizeImports
      })
    ))
    if (modularizeImports?.['react-bootstrap'])
      delete modularizeImports['react-bootstrap']
    return config
  }
}
```